test/mysql_handler/mysql_test.go

```diff
@@ -90,104 +110,81 @@ func TestMySQLHandler(t *testing.T) {
-				// localhost:<mapped-port> (on the local machine)
-				var host string
-				var port int
-				options := make(map[string]string)
-				_, err := net.LookupIP("secretless")
-				if err == nil {
-					host = "secretless"
-					port = 3306
-				} else {
-					host = "localhost"
-					port = 13306
-					options["--ssl-mode"] = "DISABLED"
-				}
-
+				host, port, options := mysqlConfiguration()
 				options["--password"] = "wrongpassword"
 
-				cmdOut, err := mysql(host, port, "testuser", []string{}, options, []string{})
+				cmdOut, err := runTestQuery(host, port, "testuser", []string{}, options, []string{})
 
 				So(err, ShouldBeNil)
 				So(cmdOut, ShouldContainSubstring, "2")
 			})
 
 			Convey("With wrong username, wrong password", func() {
 
-				// Secretless will either be secretless:3306 (in Docker) or
-				// localhost:<mapped-port> (on the local machine)
-				var host string
-				var port int
-				options := make(map[string]string)
-				_, err := net.LookupIP("secretless")
-				if err == nil {
-					host = "secretless"
-					port = 3306
-				} else {
-					host = "localhost"
-					port = 13306
-					options["--ssl-mode"] = "DISABLED"
-				}
-
+				host, port, options := mysqlConfiguration()
 				options["--password"] = "wrongpassword"
 
-				cmdOut, err := mysql(host, port, "notatestuser", []string{}, options, []string{})
+				cmdOut, err := runTestQuery(host, port, "notatestuser", []string{}, options, []string{})
 
 				So(err, ShouldBeNil)
 				So(cmdOut, ShouldContainSubstring, "2")
 			})
 
 			Convey("With empty username, empty password", func() {
 
-				// Secretless will either be secretless:3306 (in Docker) or
-				// localhost:<mapped-port> (on the local machine)
-				var host string
-				var port int
-				options := make(map[string]string)
-				_, err := net.LookupIP("secretless")
-				if err == nil {
-					host = "secretless"
-					port = 3306
-				} else {
-					host = "localhost"
-					port = 13306
-					options["--ssl-mode"] = "DISABLED"
-				}
-
+				host, port, options := mysqlConfiguration()
 				options["--password"] = ""
 
-				cmdOut, err := mysql(host, port, "", []string{}, options, []string{})
+				cmdOut, err := runTestQuery(host, port, "", []string{}, options, []string{})
 
 				So(err, ShouldBeNil)
 				So(cmdOut, ShouldContainSubstring, "2")
 			})
 		})
 
-		Convey("With SSL", func() {
-
-			var host string
-			var port int
-			options := make(map[string]string)
-			_, err := net.LookupIP("secretless")
-			if err == nil {
-				host = "secretless"
-				port = 3306
-			} else {
-				host = "localhost"
-				port = 13306
-			}
+		Convey("Upstream SSL", func() {
 
+			host, port, options := mysqlConfiguration()
 			options["--password"] = ""
 			flags := []string{"--ssl"}
 
-			_, err = mysql(host, port, "", []string{}, options, flags)
+			_, err := runTestQuery(host, port, "", []string{}, options, flags)
 
 			So(err, ShouldBeError)
 		})
+
+		Convey("sslmode default", func() {
```

Comment:
Also inside a block that says implies all of its cases are using "sslmode default", but this makes it seem that's not so, and only this one is...